File: modules/01-basic-syntax/go/types_inference.go
package main

import (
	"fmt"
)

func takes_u32(x uint32) {
  fmt.Printf("u32: %d\n", x)
}

func takes_i8(y int8) {
  fmt.Printf("i8: %d\n", y)
}

// Types inference
func main() {
  fmt.Println("------------------------\n")

  x := 10
  y := 20

  takes_u32(x)
  takes_i8(y)
  // takes_u32(y)
}

File: modules/01-basic-syntax/typescript/types_inference.ts
function takes_u32(x: number) {
  console.log(`u32: ${x}`);
}

function takes_i8(y: number) {
  console.log(`i8: ${y}`);
}

// Types inference
console.log("------------------------\n");

let x = 10;
let y = 20;

takes_u32(x);
takes_i8(y);
takes_u32(y);
